django/contrib/admin/checks.py

```diff
@@ -105,6 +105,14 @@ def check_dependencies(**kwargs):
                 "the admin application.",
                 id='admin.E404',
             ))
+        if ('django.template.context_processors.request'
+                not in django_templates_instance.context_processors):
+            errors.append(checks.Warning(
+                "'django.template.context_processors.request' must be enabled "
+                "in DjangoTemplates (TEMPLATES) in order to use the admin "
+                "application. In Django 4.0, this will be an error.",
+                id='admin.E411',
+            ))
```

Comment:
I think we should add this check only if `enable_nav_sidebar` is enabled on any site and raise an error instead of warning :thinking:

Reply:
This could work: it's only the `current-app`, `current-model`, `current-page` attribute values that are dependent on `request`, and for the index page app list they're always not set anyway, so it's only when the side bar is set that `request` is serving any purpose.

Reply:
I'm a bit unsure about this,

1. The `enable_nav_sidebar` is `True` by default. So making it an error is backward incompatible. Users that had a working `settings.py` in Django 3.0, will now receive a system check failure.

2. While `request` is only limited to the sidebar at the moment, I think it is very easy to slip it into other templates without noticing an issue. After all, it went into the sidebar without notice. My concern is that it will provide a false sense of correctness overtime and it would be safer to always require the context processor.

Reply:
I was thinking about leaving it as a warning and disabling the sidebar when `request` is not provided, so about a soft requirement:

- raise a warning (that will remain a warning in the future) if `enable_nav_sidebar`, e.g. 
  ```python
  checks.Warning(
     "'django.template.context_processors.request' must be enabled "
     "in DjangoTemplates (TEMPLATES) in order to use the navigation "
     "sidebar in the admin application."
     id='admin.W411',
  )
  ```
- disable a navigation sidebar (even in a template) when `request` is not provided.

Reply:
My thought overnight was just to have a warning and leave it at that. 

The sidebar still works without the request: we're just missing the CSS highlight for the current model, and an aria-role. 

98% of projects have request context in play already. A warning is enough for the 2%. Discuss.

Reply:
Agreed

Reply:
OK. I'll adjust this now on that basis, so we can get it in this morning. Thanks both!